src/plot_api/plot_api.js

```diff
@@ -2219,7 +2219,7 @@ Plotly.animate = function(gd, frameOrGroupNameOrFrameList, animationOpts) {
 
                 var nextFrame = {
                     frame: computedFrame,
-                    name: frameList[i].name,
+                    name: frameList[i].name || computedFrame.name,
```

Comment:
No need to move it. I was just curious.

How hard would it to be to add a test for that case? If not so hard, then let's add one,